include/heuristic/HeuristicMapper.hpp

```diff
@@ -93,13 +103,45 @@ class HeuristicMapper : public Mapper {
       } else {
         throw QMAPException("Something wrong in applySWAP.");
       }
+
+      if (considerFidelity) {
+        // accounting for fidelity difference of single qubit gates (two qubit
+        // gates are handled in the heuristic)
+        costFixed +=
+            (-(singleQubitGateMultiplicity.at(static_cast<std::size_t>(q2)) -
+               singleQubitGateMultiplicity.at(static_cast<std::size_t>(q1))) *
+                 log2(singleQubitFidelities.at(swap.first)) -
```

Comment:
Would it be helpful to straight up store log-fidelities? Avoids the constant computation here.

Reply:
Well it would introduce the need for 2 additional member data structures inside `Architecture` (since raw fidelity values are already in use in some methods of `Architecture` itself, so existing `singleQubitFidelities` and `fidelityTable` cannot just replaced with log versions). 
But considering the number of calls to log2 here, I think you are right. Will also be implemented in a following commit on this PR. 
Thank you

Reply:
These additional data structures should probably remain really small even for larger architectures, so I think their overhead is justified.
If that turns out to not be the case, the occurrences where the fidelities are used in a non-log fashion could probably be refactored.